src/Site/SiteManager.py

```diff
@@ -24,15 +25,24 @@ def load(self):
             self.sites = {}
         address_found = []
         added = 0
+        serving = 0
         # Load new adresses
         for address in json.load(open("%s/sites.json" % config.data_dir)):
             if address not in self.sites and os.path.isfile("%s/%s/content.json" % (config.data_dir, address)):
                 s = time.time()
                 self.sites[address] = Site(address)
                 logging.debug("Loaded site %s in %.3fs" % (address, time.time() - s))
                 added += 1
+                if self.sites[address].settings["serving"]:
+                    serving += 1
             address_found.append(address)
 
+        # check if there are enough onions available
+        if sys.modules.get("main") and sys.modules["main"].file_server:
+            tor_manager = sys.modules["main"].file_server.tor_manager
+            if tor_manager and tor_manager.numOnions() < serving+1:
```

Comment:
This is unpractical. `serving`  here refers to the number of sites visited, not hosted. I would remove that condition completely.